test/Common.FunctionalTests/AppOfflineTests.cs

```diff
@@ -29,32 +30,114 @@ public class AppOfflineTests : IISFunctionalTestBase
             AddAppOffline(deploymentResult.DeploymentResult.ContentRoot);
 
             await AssertAppOffline(deploymentResult);
+            AssertFilesNotLocked(deploymentResult);
         }
 
-        [ConditionalTheory]
-        [InlineData(HostingModel.InProcess)]
-        [InlineData(HostingModel.OutOfProcess)]
-        public async Task AppOfflineDroppedWhileSiteIsDown_CustomResponse(HostingModel hostingModel)
+        [ConditionalFact]
+        public async Task AppOfflineDroppedWhileSiteFailedToStartInShim_AppOfflineServed_InProcess()
         {
-            var expectedResponse = "The app is offline.";
-            var deploymentResult = await DeployApp(hostingModel);
+            var deploymentResult = await DeployApp(HostingModel.InProcess);
+
+            Helpers.ModifyAspNetCoreSectionInWebConfig(deploymentResult, "processPath", "nonexistent");
+
+            var result = await deploymentResult.HttpClient.GetAsync("/");
+            Assert.Equal(500, (int)result.StatusCode);
+            Assert.Contains("500.0", await result.Content.ReadAsStringAsync());
+
+            AddAppOffline(deploymentResult.DeploymentResult.ContentRoot);
+
+            await AssertAppOffline(deploymentResult);
+            AssertFilesNotLocked(deploymentResult);
+        }
+
+        [ConditionalFact]
+        [RequiresIIS(IISCapability.ShutdownToken)]
+        public async Task AppOfflineDroppedWhileSiteFailedToStartInRequestHandler_SiteStops_InProcess()
+        {
+            var deploymentResult = await DeployApp(HostingModel.InProcess);
 
-            AddAppOffline(deploymentResult.DeploymentResult.ContentRoot, expectedResponse);
+            // Set file content to empty so it fails at runtime
+            File.WriteAllText(Path.Combine(deploymentResult.DeploymentResult.ContentRoot, "Microsoft.AspNetCore.Server.IIS.dll"), "");
 
-            await AssertAppOffline(deploymentResult, expectedResponse);
+            var result = await deploymentResult.HttpClient.GetAsync("/");
+            Assert.Equal(500, (int)result.StatusCode);
+            Assert.Contains("500.30", await result.Content.ReadAsStringAsync());
+
+            AddAppOffline(deploymentResult.DeploymentResult.ContentRoot);
+            AssertStopsProcess(deploymentResult);
+        }
+
+        
+        [ConditionalFact]
+        [RequiresIIS(IISCapability.ShutdownToken)]
+        public async Task AppOfflineDroppedWhileSiteFailedToStart_SiteStops_OutOfProcess()
```

Comment:
First, I think the test name should contain `_AppOfflineServed_` instead of `_SiteStops`, based on the actual implementation of the test.

Second, should this test be combined with `AppOfflineDroppedWhileSiteFailedToStartInShim_AppOfflineServed_InProcess` as a conditional theory with `HostingModel` parameters?  The tests are very similar.

Reply:
Will do.